libp2p/protocols/secure/secure.nim

```diff
@@ -29,7 +29,7 @@ method writeMessage*(c: SecureConn, data: seq[byte]) {.async, base.} =
 
 method handshake(s: Secure,
                  conn: Connection,
-                 initiator: bool = false): Future[SecureConn] {.async, base.} =
+                 initiator: bool): Future[SecureConn] {.async, base.} =
```

Comment:
why remove the default?

Reply:
why have a default? for every intiiator, isn't there a non-initiator? why is one more important than the opther?

Reply:
simply cos defaults are a bad habit and dangerous generally, specially here had no use anymore

Reply:
> why have a default? for every intiiator, isn't there a non-initiator? why is one more important than the opther?

Historically, `secure` was called to initiate the flow and there was no flag to indicate who is the initiator because `secure` already implied it. The distinction came with Noise, the default was there to not break/change the public API. 

Why does Noise have this requirement and are we signaling it properly in all the right places?

Reply:
The reasoning behind the default is that not all securing mechanisms need this flag, for example, secio doesn't use it.

> simply cos defaults are a bad habit and dangerous generally, specially here had no use anymore

This is a bit of a generalization. Every tool has it's use and not everything is a nail that needs a hammer - https://en.wikipedia.org/wiki/Law_of_the_instrument

Reply:
The reason for Noise is handshake patterns are different between initiator/receiver:
https://noiseprotocol.org/noise.html#handshake-patterns

Reply:
> The reasoning behind the default is that not all securing mechanisms need this flag, for example, secio doesn't use it.

This is an argument to not have a default, right? I mean, if I'm coming from `secio`, I would want to know about this important difference between the two protocols, or my application won't work and I'll have to spend time looking for the cause - finding a default boolean is hard.

Basically, there's no meaningful way to use the protocol with _only_ the default setting.